### ISSUE TITLE
Rails 6 Single Table Inheritance

### ISSUE THREAD

**Issue by clemens1483** [zdennis/activerecord-import#735]
Version: `1.1.0`
Rails: `6.0.4`

**Issue**
Given a table with single table inheritance
```ruby
  create_table "humans", id: :integer, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci", force: :cascade do |t|
    t.string "type", null: false
    t.string "name", null: false
  end
```

And the following models
```ruby
class Human < ApplicationRecord
end

class Child < Human
end

class Parent < Human
end
```

When I import rows for one of the inherited models
```ruby
Parent.import([{name: "Blah"}])
```

Then activerecord_import does not populate the `type` field according to the class and an exception is raised
```
INSERT INTO `humans` (`name`) VALUES ('Blah')

ActiveRecord::NotNullViolation: Mysql2::Error: Field 'type' doesn't have a default value
```

**Expected Behavior**
Then activerecord_import populates the `type` field according to the class
```
INSERT INTO `humans` (`name`, `type`) VALUES ('Blah', 'Parent')
```

When using the gem with rails `5.2`, activerecord import automatically populated the type column.

**Comment by jkowens** [comment]
Thanks for reporting the issue 👍

**Comment by clemens1483** [comment]
After doing some digging I think I found the root of the problem.

In `ActiveRecord::Base#import_without_validations_or_callbacks`, `#scope_attributes` is used. This method returned the sti column in rails 5.2 but just returns an empty hash (provided no other scopes are used) in rails 6.0. The cause for this seems to have been this pr https://github.com/rails/rails/pull/39967.

```ruby
# rails 5
Parent.scope_attributes # => {"type"=>"Human"}

# rails 6
Parent.scope_attributes # => {}
```


```ruby
# lib/activerecord-import/import.rb
    def import_without_validations_or_callbacks
      ...
      scope_columns, scope_values = scope_attributes.to_a.transpose

      unless scope_columns.blank?
        scope_columns.zip(scope_values).each do |name, value|
          name_as_sym = name.to_sym
          next if column_names.include?(name_as_sym)

          is_sti = (name_as_sym == inheritance_column.to_sym && self < base_class)
          value = Array(value).first if is_sti

          column_names << name_as_sym
          array_of_attributes.each { |attrs| attrs << value }
        end
      end

      ....
```
Since `scope_attributes` now returns `{}` for sti tables, `scope_columns` is nil so that `scope_columns.blank?` returns true and the block is never executed and we never get to `is_sti = (name_as_sym == inheritance_column.to_sym && self < base_class)` causing the sti column not to be added to the column names.

The issue might be fixed (this is just a suggestion and entirely untested) by using something like
```ruby
if finder_needs_type_condition?
  column_names << inheritance_column.to_sym
  array_of_attributes.each { |attrs| attrs << sti_name }
end
```

**Comment by jkowens** [comment]
@clemens1483 your proposed solution looks like it would be a great update. I'll have to carve out some time to test it out.

**Comment by jkowens** [comment]
@clemens1483 can you test out the master branch to make sure it works as expected for you now?

**Comment by jkowens** [comment]
Released in v1.2.0 🎉
